kotlin/samples/dungeon/common/src/main/java/com/squareup/sample/dungeon/GameWorkflow.kt

```diff
@@ -156,16 +154,12 @@ class GameWorkflow(
         )
 
         // Check if AI captured player.
-        return@runningWorker if (newGame.isPlayerEaten) {
-          enterState(
-              state.copy(game = newGame),
-              emittingOutput = PlayerWasEaten
-          )
-        } else {
-          enterState(
-              state.copy(game = newGame),
-              emittingOutput = output
-          )
+        return@runningWorker if (newGame.isPlayerEaten) WorkflowAction {
```

Comment:
Another spot to consider moving this all out into a single bigger action method.

Reply:
Backing away from this one. Its `tick` lambda leans heavily on the child renderings.